dep/efsw/CMakeLists.txt

```diff
@@ -1,88 +1,112 @@
+            src/efsw/platform/win/ThreadImpl.cpp)
+  else ()
     list(APPEND SRCS
-      src/efsw/platform/posix/FileSystemImpl.cpp
-      src/efsw/platform/posix/MutexImpl.cpp
-      src/efsw/platform/posix/SystemImpl.cpp
-      src/efsw/platform/posix/ThreadImpl.cpp)
-  endif()
+            src/efsw/platform/posix/FileSystemImpl.cpp
+            src/efsw/platform/posix/MutexImpl.cpp
+            src/efsw/platform/posix/SystemImpl.cpp
+            src/efsw/platform/posix/ThreadImpl.cpp)
+  endif ()
 
-  if (WIN32)
-    list(APPEND SRCS
-      src/efsw/WatcherWin32.cpp
-      src/efsw/FileWatcherWin32.cpp)
+  if (APPLE)
+    list(REMOVE_ITEM SRCS
+            "src/efsw/WatcherInotify.cpp"
+            "src/efsw/WatcherWin32.cpp"
+            "src/efsw/FileWatcherInotify.cpp"
+            "src/efsw/FileWatcherWin32.cpp")
+
+    exec_program(uname ARGS -v OUTPUT_VARIABLE OSX_VERSION)
+    string(REGEX MATCH "[0-9]+" OSX_VERSION ${OSX_VERSION})
+    if (NOT OSX_VERSION GREATER 9)
+      add_definitions(-DEFSW_FSEVENTS_NOT_SUPPORTED)
+    endif ()
+  elseif (WIN32)
+    list(REMOVE_ITEM SRCS
+            "src/efsw/WatcherKqueue.cpp"
+            "src/efsw/WatcherFSEvents.cpp"
+            "src/efsw/WatcherInotify.cpp"
+            "src/efsw/FileWatcherKqueue.cpp"
+            "src/efsw/FileWatcherInotify.cpp"
+            "src/efsw/FileWatcherFSEvents.cpp")
   elseif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    list(APPEND SRCS
-      src/efsw/FileWatcherInotify.cpp
-      src/efsw/WatcherInotify.cpp)
+    list(REMOVE_ITEM SRCS
+            "src/efsw/WatcherKqueue.cpp"
+            "src/efsw/WatcherFSEvents.cpp"
+            "src/efsw/WatcherWin32.cpp"
+            "src/efsw/FileWatcherKqueue.cpp"
+            "src/efsw/FileWatcherWin32.cpp"
+            "src/efsw/FileWatcherFSEvents.cpp")
 
     if (NOT EXISTS "/usr/include/sys/inotify.h" AND NOT EXISTS "/usr/local/include/sys/inotify.h")
       add_definitions(-DEFSW_INOTIFY_NOSYS)
-    endif()
-  elseif (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD" OR APPLE)
-    list(APPEND SRCS
-      src/efsw/FileWatcherKqueue.cpp
-      src/efsw/WatcherKqueue.cpp)
-
-    if (APPLE)
-      list(APPEND SRCS
-        src/efsw/FileWatcherFSEvents.cpp
-        src/efsw/WatcherFSEvents.cpp)
-
-      exec_program(uname ARGS -v  OUTPUT_VARIABLE OSX_VERSION)
-      string(REGEX MATCH "[0-9]+" OSX_VERSION ${OSX_VERSION})
-      if (NOT OSX_VERSION GREATER 9)
-        add_definitions(-DEFSW_FSEVENTS_NOT_SUPPORTED)
-      endif()
+    endif ()
+  elseif (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    list(REMOVE_ITEM SRCS
+            "src/efsw/WatcherInotify.cpp"
+            "src/efsw/WatcherWin32.cpp"
+            "src/efsw/WatcherFSEvents.cpp"
+            "src/efsw/FileWatcherInotify.cpp"
+            "src/efsw/FileWatcherWin32.cpp"
+            "src/efsw/FileWatcherFSEvents.cpp")
+  endif ()
 
-      set(OPTIONAL_MAC_LINK_LIBRARIES "-framework CoreFoundation" "-framework CoreServices")
-    endif()
-  endif()
+  if (MSVC)
```

Comment:
While it is part of original CMakeLists.txt on bitbucket, you need to remove this part - we don't want to have warnings (or any compile flags actually) messed with in /dep